CONTRIBUTING.md

```diff
@@ -1,25 +1,116 @@
+
+- Your issue will be catagorised as a T-Defect, T-Enhancement or T-Documentation by the maintainer team.
+- You *may* have a severity assigned as an estimate to understand how bad the problem is and how many users are affected.
+
+The assigned labels are at the maintainers discretion, and we will make every effort to be transparent when triaging. We do
+not, as a rule, assign priority labeling to issues.
+
+## Contributing documentation
+
+Documentation is important to us, as bridges are complex beasts and rely on good documentation for both
+administrators and users. When writing documentation improvements, have these
+ 
+ - Our documentation is currently English, though eventually we might be able to adopt a multi-lingual documentation system.
+ - Please take care to proofread your documentation, and where possible avoid fluff.
+
+## Contributing code
+
+First of all, thank you for considering making a change to one of our projects. Every change no matter the size makes a difference! 
+
+### 🖌️ Code style
+
+Each repository contains an `eslint` configuration which will dictate the style of the code. All code should be written in TypeScript. At time of
+writing, we target ES2020 (the latest version supported by Node 14). The CI will lint your code automatically, but you can save yourself some time by running (`yarn lint`/`npm lint`) to check locally.
+
+### 🧪 Tests / CI
+
+To test your changes, you can run the `test` command with either `yarn test` or `npm test`. Some projects may have additional testing constraints
+noted in the project-specific section below.
+
+Please be aware that reviewers will expect CI to be passing before your changes will be approved, and priority will be given to PRs that
+pass CI when reviewing too. If you can't get the CI to pass, please reach out to us (and do not assume that it's always your change that broke it!).
+
+**As a rule, code does not get merged onto the `develop` branch without all CI tests passing.**
+
+### Tips for good quality submissions
+
+ - When writing new features, remember to document this in the repositories chosen documentation system.
+ - PRs should aim to be as constrained as possible: Do not attempt to solve multiple isolated issues in a single PR.
+   - A good indication is that your changelog entry contains multiple statements. That usually means you need to consider splitting up your PR :)
+ - It's totally okay to submit draft PRs with the intention of getting feedback. Please use the GitHub comments feature to comment on lines you would like assistance with.
+ - Avoid writing TODOs / XXX comments in code. If you must, create an issue first with the details and link to it in the code.
+
+
+### ⬇️ Pull Requests
+
+When making a pull request, please be aware of these best practises.
+
+- Targets `develop` (unless it explicitly depends upon another feature, then depend on that branch and comment to that effect in the PR body).
```

Comment:
We should probably head towards using either `develop` or `main` as the standard branch across all projects, but in the meantime let's call out divergences in the contributing file (I think it might just be hookshot, anyway).